rocky/tools/models.py

```diff
@@ -170,6 +185,55 @@ class STATUSES(models.TextChoices):
     def blocked(self):
         return self.status == OrganizationMember.STATUSES.BLOCKED
 
+    @cached_property
+    def is_admin(self) -> bool:
+        return self.groups.filter(name=GROUP_ADMIN).exists()
+
+    @cached_property
+    def is_redteam(self) -> bool:
+        return self.groups.filter(name=GROUP_REDTEAM).exists()
+
+    @cached_property
+    def is_client(self) -> bool:
+        return self.groups.filter(name=GROUP_CLIENT).exists()
+
+    def get_permissions_codenames(self, permissions: Union[str, tuple] = None) -> List[str]:
+        codenames = []
+
+        if permissions is None:
+            raise ValueError("Permission(s) not defined")
+        if isinstance(permissions, str):
+            perms = (permissions,)
+        else:
+            perms = permissions
+        for perm in perms:
+            if perm.count(".") == 1:
+                app_label, codename = perm.split(".")
+                if app_label and codename:
+                    if ContentType.objects.filter(app_label=app_label, model="organization").exists():
+                        codenames.append(codename)
+            else:
+                raise ValueError("Permissions should be defined as: 'app_label.codename'.")
+        return codenames
+
+    def has_member_perms(self, permissions: Union[str, tuple] = None) -> bool:
+        if self.user.is_superuser and self.user.is_active and self.status is not OrganizationMember.STATUSES.BLOCKED:
+            return True
+        codenames = self.get_permissions_codenames(permissions)
+        has_permissions = []
+
+        for codename in codenames:
+            for group in self.groups.all():
+                group_permissions_codenames = group.permissions.values_list("codename", flat=True)
```

Comment:
This is wrong. Both the app and codename should match. The permission "app1.can_do_something" is not the same as a permission "app2.can_do_something". It also does too many queries.

We can get all the permissions with a single query:

```
perms = ["%s.%s" % (ct, name) for ct, name in Permission.objects.filter(group__organizationmember=member).values_list("content_type__app_label", "codename")]
```